pkg/manifests/manifests.go

```diff
@@ -1659,6 +1662,30 @@ func (f *Factory) PrometheusK8s(grpcTLS *v1.Secret, trustedCABundleCM *v1.Config
 	return p, nil
 }
 
+func (f *Factory) queryLogFileConfig(p *monv1.Prometheus, queryLogFile string) (string, error) {
+	if strings.HasPrefix(queryLogFile, "/") && !(strings.HasPrefix(queryLogFile, "/dev") || strings.HasPrefix(queryLogFile, "/prometheus")) {
+		if strings.Count(queryLogFile, "/") == 1 {
+			return "", errors.Wrap(ErrConfigValidation, "attempting to mount query log file on root \"/\"")
+		}
```

Comment:
nit:

Refactor a bit for readability :)

```suggestion
                dirPath := filepath.Dir(queryLogFile)
		if dirPath == "/" {
		  return "", errors.Wrap(ErrConfigValidation, `attempting to mount query log file on root "/"`)
		}
		if dirPath == "/prometheus" || dirPath = "/dev"{
		  return queryLogFile, nil
		}
		... volume mount ..
```